include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -0,0 +1,221 @@
+            <
+                Linear
+            >::type::units linear_units_type;
+
+        BOOST_ASSERT(geometry::num_points(linear) != 0);
+
+        typedef longitude_interval<calculation_type> interval_type;
+
+        calculation_type const pi = math::pi<calculation_type>();
+
+        std::vector<interval_type> longitude_intervals;
+        std::back_insert_iterator
+            <
+                std::vector<interval_type>
+            > oit(longitude_intervals);
+
+        calculation_type lat_min = 0, lat_max = 0;
+        bool first = true;
+        for (iterator_type seg_it = geometry::segments_begin(linear);
+             seg_it != geometry::segments_end(linear);
+             ++seg_it, first = false)
+        {
+            typename point_type<Linear>::type p[2];
+            detail::assign_point_from_index<0>(*seg_it, p[0]);
+            detail::assign_point_from_index<1>(*seg_it, p[1]);
+
+            calculation_type lon1 = geometry::get<0>(p[0]);
+            calculation_type lat1 = geometry::get<1>(p[0]);
+            calculation_type lon2 = geometry::get<0>(p[1]);
+            calculation_type lat2 = geometry::get<1>(p[1]);
+
+            math::normalize_spheroidal_coordinates
+                <
+                    linear_units_type
+                >(lon1, lat1);
+
+            math::normalize_spheroidal_coordinates
+                <
+                    linear_units_type
+                >(lon2, lat2);
+
+            math::convert_coordinates<linear_units_type, radian>(lon1, lat1);
+            math::convert_coordinates<linear_units_type, radian>(lon2, lat2);
+
+            compute_mbr_of_segment::apply(lon1, lat1, lon2, lat2); 
+            oit = push_interval(lon1, lon2, oit);
+
+            if (first)
+            {
+                lat_min = lat1;
+                lat_max = lat2;
+            }
+
+            // update min and max latitude, if needed
+            if (math::smaller(lat1, lat_min))
+            {
+                lat_min = lat1;
+            }
+
+            if (math::larger(lat2, lat_max))
+            {
+                lat_max = lat2;
+            }
+        }
+
+        calculation_type lon_min = 0, lon_max = 0;
+        envelope_range_of_longitudes
+            <
+                radian
+            >::apply(longitude_intervals, lon_min, lon_max);
+
+        math::convert_coordinates
+            <
+                radian, typename coordinate_system<Box>::type::units
+            >(lon_min, lat_min);
+
+        math::convert_coordinates
+            <
+                radian, typename coordinate_system<Box>::type::units
+            >(lon_max, lat_max);
+
+        assign_values(mbr,
+                      boost::numeric_cast<box_coordinate_type>(lon_min),
+                      boost::numeric_cast<box_coordinate_type>(lat_min),
+                      boost::numeric_cast<box_coordinate_type>(lon_max),
+                      boost::numeric_cast<box_coordinate_type>(lat_max));
+    }
```

Comment:
I think the reason I did not use `set_from_radian` is that:

1. I think it does not work with boxes (it takes only one index template parameter, and most probably we should change that...)
2. I think it does not do any conversion *or* the conversion done is implicit, which I wanted to avoid.

In any case I will check again if it can/should be done the way you suggest,